src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -3354,7 +3354,8 @@ public static int SplitAny(this ReadOnlySpan<char> source, Span<Range> destinati
 
             // If the separators list is empty, whitespace is used as separators.  In that case, we want to ignore TrimEntries if specified,
             // since TrimEntries also impacts whitespace.
-            if (separators.IsEmpty)
+            // The TrimEntries flag must be left intact if we are constrained by count because we need to process last substring.
+            if (separators.IsEmpty && destination.Length > source.Length)
```

Comment:
Why > rather than >= ?  Is there a corner-case that fails if >= is used?

Reply:
I used > because in tests was `new Range[value.Length + 1]` for count == int.MaxValue.

I did some testing and >= is fine. `destination.Length >= source.Length - 1` is first to break corner-case I could think of.

Reply:
Should I replace > with >=?

Reply:
You can leave it as >.  It's already likely a large overestimate, and it's easier to explain the reasoning that you can't possibly have N+1 segments for N characters.